app/src/main/java/me/ccrama/redditslide/util/LinkUtil.java

```diff
@@ -107,37 +114,37 @@ public static void openCustomTab(@NonNull String url, int color,
     }
 
     public static void openUrl(@NonNull String url, int color, @NonNull Activity contextActivity,
-            int adapterPosition, Submission submission) {
-        if (!SettingValues.web) {
-            // External browser
+            @Nullable Integer adapterPosition, @Nullable Submission submission) {
+        if (!(contextActivity instanceof ReaderMode) && ((SettingValues.readerMode
```

Comment:
Analyze this entire condition for deciding to open `ReaderMode` and let me know if it's right.

Reply:
"It's not! You'll never enter reader mode if you have night mode on!"
Thanks, other Piwp.

Reply:
This entire condition may be valuable to move somewhere where it's more reliable to call out to since it's a double check.
But now I am just talking to myself so that's up to you @ccrama